### PR TITLE
ci: Orb tools 11 migration

### PR DESCRIPTION
**Pull request by brivu**
## This PR includes

- Adding new AWS Context for testing
- Creating new DynamoDB tables for testing
- Creating new S3 buckets for testing
- Updating orb required orb versions 
- Additional retry if `terraform destroy` command times out

### CODE REVIEW

File: .circleci/config.yml.bak [hunk 1]
```diff
@@ -0,0 +1,726 @@
+version: 2.1
```
Comment (KyleTryon) [inline .circleci/config.yml.bak:1]: Delete this file

File: .circleci/test-deploy.yml [hunk 2]
```diff
@@ -0,0 +1,688 @@
...
+          context: [CPE_ORBS_AWS]
+      - orb-tools/pack:
+          filters: *filters
+      - orb-tools/publish:
+          orb-name: circleci/aws-ecs
+          vcs-type: << pipeline.project.type >>
+          pub-type: production
+          requires:
+            - orb-tools/pack
+            # - ec2_tear-down-test-env
```
Comment (KyleTryon) [inline .circleci/test-deploy.yml:591]: Are these meant to be commented out?

Comment (KyleTryon) [reply .circleci/test-deploy.yml:591]: Are all jobs tested prior to publishing?

File: .circleci/test-deploy.yml [hunk 3]
```diff
@@ -0,0 +1,688 @@
...
+      - tear-down-test-env:
+          name: codedeploy_fargate_tear-down-test-env
+          requires:
+            - codedeploy_fargate_test-update-and-wait-service-job
+          terraform-image: "hashicorp/terraform:1.1.9"
+          aws-resource-name-prefix: ${AWS_RESOURCE_NAME_PREFIX_CODEDEPLOY_FARGATE}
+          terraform-config-dir: "tests/terraform_setup/fargate_codedeploy"
+          context: [CPE_ORBS_AWS]
+      - orb-tools/pack:
+          filters: *filters
```
Comment (KyleTryon) [inline .circleci/test-deploy.yml:584]: Make sure all jobs in the workflow that will be required by `publish` at any point down stream, also have the `filters` key and anchor value

**Review by KyleTryon — CHANGES_REQUESTED** [review]

File: src/jobs/update-task-definition-from-json.yml [hunk 4]
```diff
@@ -1,3 +1,6 @@
+description: |
+  Install AWS CLI and  a task definition from a json file.
```
[the remark this answers is not in the document]
Comment (brivu) [reply src/jobs/update-task-definition-from-json.yml:2]: Deleted the extra space

File: .circleci/config.yml.bak (hunk 1 above)
Comment (brivu) [reply .circleci/config.yml.bak:1]: File deleted

File: .circleci/test-deploy.yml [hunk 5]
```diff
@@ -0,0 +1,692 @@
...
+          docker-image-name: "${AWS_RESOURCE_NAME_PREFIX_EC2}:${CIRCLE_SHA1}"
+          context: [CPE_ORBS_AWS]
+      - test-task-definition-update:
+          name: ec2_test-task-definition-update
+          family-name: "${AWS_RESOURCE_NAME_PREFIX_EC2}-family"
+          context: [CPE_ORBS_AWS]
+          requires:
+            - ec2_test-update-service-command
+      - aws-ecs/deploy-service-update:
+          name: ec2_test-update-service-job
```
Comment (KyleTryon) [inline .circleci/test-deploy.yml:460]: `ec2_test-update-service-job` is a dependent of `ec2_tear-down-test-env` which has filters, so it too must have filters. All downstream jobs must have filters.

**Review by KyleTryon — CHANGES_REQUESTED** [review]

File: .circleci/test-deploy.yml (hunk 5 above)
Comment (brivu) [reply .circleci/test-deploy.yml:460]: Added filters to all downstream jobs

**Pull request merged**